src/connection_string.ts

```diff
@@ -507,6 +500,36 @@ export function parseOptions(
     );
   }
 
+  const loggerFeatureFlag = Symbol.for('@@mdb.enableMongoLogger');
+  mongoOptions[loggerFeatureFlag] = mongoOptions[loggerFeatureFlag] ?? false;
+
+  if (mongoOptions[loggerFeatureFlag]) {
+    const loggerEnvOptions: MongoLoggerEnvOptions = {
+      MONGODB_LOG_COMMAND: process.env.MONGODB_LOG_COMMAND,
+      MONGODB_LOG_TOPOLOGY: process.env.MONGODB_LOG_TOPOLOGY,
+      MONGODB_LOG_SERVER_SELECTION: process.env.MONGODB_LOG_SERVER_SELECTION,
+      MONGODB_LOG_CONNECTION: process.env.MONGODB_LOG_CONNECTION,
+      MONGODB_LOG_ALL: process.env.MONGODB_LOG_ALL,
+      MONGODB_LOG_MAX_DOCUMENT_LENGTH: process.env.MONGODB_LOG_MAX_DOCUMENT_LENGTH,
+      MONGODB_LOG_PATH: process.env.MONGODB_LOG_PATH
+    };
+    const loggerMongoClientOptions: MongoLoggerMongoClientOptions = {
+      mongodbLogPath: mongoOptions.mongodbLogPath
+    };
+    const loggerOptions = MongoLogger.resolveOptions(loggerEnvOptions, loggerMongoClientOptions);
+    const loggingComponents = [
+      loggerOptions.command,
+      loggerOptions.topology,
+      loggerOptions.serverSelection,
+      loggerOptions.connection,
+      loggerOptions.defaultSeverity
+    ];
+
+    if (loggingComponents.some(severity => severity !== SeverityLevel.OFF)) {
+      mongoOptions.mongoLogger = new MongoLogger(loggerOptions);
```

Comment:
why do we set the logger on the mongoOption at all? the instantiation should really be a part of the client instantiation, not options parsing; if we really want, we can add the parsed logger options to the mongo options; the point of the options is to provide a reference to the settings that went into configuring the client